netket/sampler/metropolis.py

```diff
@@ -421,6 +431,18 @@ def _sample_next(sampler, machine, parameters, state):
         itself, because `sample_next` contains some common logic.
         """
 
+        if sampler.chunk_size is None:
+
+            def apply_machine(σ):
+                return machine.apply(parameters, σ)
+
+        else:
+
+            def apply_machine(σ):
+                σ = σ.reshape(-1, sampler.chunk_size, σ.shape[-1])
+                f = lambda s: machine.apply(parameters, s)
+                return jax.lax.map(f, σ).ravel()
```

Comment:
@inailuig will this preserve sharding information correctly, or should we do a `shard_map` dance?

Reply:
Where does apply_chunked live? I'm happy to use it if it already exists

Reply:
I think I found it. Please check if it's correctly used! @inailuig

Reply:
yeah those tests are broken on CI, but running locally they pass. I believe it's because we get out of memory on the GitHub runner. Anyhow this is not a problem.